network/src/logging.rs

```diff
@@ -10,37 +10,60 @@
 //! use network::logging::*;
 //!
 //! info!(
-//!   network_log(network_events::CONNECTIVITY_MANAGER_LOOP, &NetworkContext::mock())
-//!     .data(network_events::TYPE, network_events::START)
-//!     .field(network_events::EVENT_ID, &5)
+//!   NetworkContextSchema {
+//!     network_context: &NetworkContext::mock()
+//!   },
+//!   field_name = "field",
+//!   "Value is {} message",
+//!   5
 //! );
 //! ```
 //!
 
 use libra_config::network_id::NetworkContext;
-use libra_logger::StructuredLogEntry;
+use libra_logger::{Schema, StructuredLogEntry};
+use libra_network_address::NetworkAddress;
+use libra_types::PeerId;
 
 /// A helper function to cut down on a bunch of repeated network struct log code
 pub fn network_log(label: &'static str, network_context: &NetworkContext) -> StructuredLogEntry {
     StructuredLogEntry::new_named("network", label)
         .field(network_events::NETWORK_CONTEXT, network_context)
 }
 
+#[derive(Schema)]
+pub struct NetworkContextSchema<'a> {
+    pub network_context: &'a NetworkContext,
+}
+
+#[derive(Schema)]
+pub struct RemotePeerSchema<'a> {
+    pub network_context: &'a NetworkContext,
+    #[schema(display)]
+    pub remote_peer: &'a PeerId,
+}
+
+#[derive(Schema)]
+pub struct RemotePeerWithAddressSchema<'a> {
+    pub network_context: &'a NetworkContext,
+    #[schema(display)]
+    pub remote_peer: &'a PeerId,
+    #[schema(display)]
+    pub network_address: &'a NetworkAddress,
+}
+
```

Comment:
This is probably personal preference, but it might make things a little easier to maintain if you have a single Schema for the whole network, defining all of the field names/types you want and use the generated setters like is done in the Saftey-rules crate. E.g:

```
struct LogSchema<'a> {
    network_context: &'a NetworkContext,
    remote_peer: Option<&'a PeerId>,
   network_address: Option<&'a NetworkAddress>,
   . . .
}
```